posthog/models/special_migration.py

```diff
@@ -0,0 +1,43 @@
+from django.db import models
+
+
+# an enum, essentially
+class MigrationStatus:
+    NotStarted = 0
+    Running = 1
+    CompletedSuccessfully = 2
+    Errored = 3
+    RolledBack = 4
+
+
+class SpecialMigration(models.Model):
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["name"], name="unique name",)]
+
+    id: models.BigAutoField = models.BigAutoField(primary_key=True)
+    name: models.CharField = models.CharField(max_length=400, null=False, blank=False)
+    progress: models.IntegerField = models.IntegerField(null=False, blank=False, default=0)
+    status: models.PositiveSmallIntegerField = models.PositiveSmallIntegerField(
+        null=False, blank=False, default=MigrationStatus.NotStarted
+    )
+
+    current_operation_index: models.PositiveSmallIntegerField = models.PositiveSmallIntegerField(
```

Comment:
Ditto

Reply:
This is very important. It helps us keep track of what op we're at. 

TL;DR:

- **Status:** one of NotStarted, Running, CompletedSuccessfully, Errored, RolledBack (and maybe more to come)
- **Progress:** Integer between 0 and 100 indicating the true progress of a migration 
- **Current op index:** _n_ where `n >=0 and n <= len(operations)`. This is _not_ real progress. Think about the example migration in the other PR. The first ops should complete in milliseconds, and so should the last few. There's one op in the middle that can take up days. This is the one column that isn't really "meta" - we actually depend on it to know what to run next, show the user where we failed, resume operations on crashes, etc.

Reply:
Ah I thought we wanted to keep the migrations sequential and with max concurrency hardcoded to 1. In this case it make sense I think